d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -1167,14 +1189,22 @@ public DarkClusterConfigMap getDarkClusterConfigMap(String clusterName) throws S
   @Override
   public void getDarkClusterConfigMap(String clusterName, Callback<DarkClusterConfigMap> callback)
   {
-    Callback<DarkClusterConfigMap> wrappedCallback = new TimeoutCallback<>(_executor, _timeout,
-            _unit, callback);
+    Callback<DarkClusterConfigMap> wrappedCallback = callback;
+    try
+    {
+      wrappedCallback = new TimeoutCallback<>(_executor, _timeout, _unit, callback);
+    }
+    catch (RejectedExecutionException e)
+    {
+      _log.debug("Executor rejected new tasks. It has shut down or its queue size has reached max limit");
+    }
+    Callback<DarkClusterConfigMap> finalWrappedCallback = wrappedCallback;
     _state.listenToCluster(clusterName, (type, name) ->
     {
       ClusterProperties clusterProperties = _state.getClusterProperties(clusterName).getProperty();
       DarkClusterConfigMap darkClusterConfigMap = clusterProperties != null ?
               clusterProperties.accessDarkClusters() : new DarkClusterConfigMap();
-      wrappedCallback.onSuccess(darkClusterConfigMap);
+      finalWrappedCallback.onSuccess(darkClusterConfigMap);
```

Comment:
How about reformat the code to following
```java
public void getDarkClusterConfigMap(String clusterName, Callback<DarkClusterConfigMap> callback) {
    try {
        Callback<DarkClusterConfigMap> wrappedCallback = new TimeoutCallback<>(_executor, _timeout, _unit, callback);
        _state.listenToCluster(clusterName, (type, name) -> {
            ClusterProperties clusterProperties = _state.getClusterProperties(clusterName).getProperty();
            DarkClusterConfigMap darkClusterConfigMap = (clusterProperties != null) ? clusterProperties.accessDarkClusters() : new DarkClusterConfigMap();
            wrappedCallback.onSuccess(darkClusterConfigMap);
        });
    } catch (RejectedExecutionException e) {
        _log.debug("Executor rejected new tasks. It has shut down or its queue size has reached max limit");
    }
}
```

Reply:
good thought, but "wrappedCallback.onSuccess" will have error "Variable used in lambda expression should be final or effectively final", so it still require making a final copy: "Callback<DarkClusterConfigMap> finalWrappedCallback = wrappedCallback;".

Reply:
I have tried the above solution. It seems worked.

Reply:
ah, I forgot to remove the "Callback<DarkClusterConfigMap> wrappedCallback = callback;" at the top, which caused the "should be final or effectively final" error. Updated as suggested. Thanks.